cargo/cargo_build_script_runner/lib.rs

```diff
@@ -141,17 +147,23 @@ impl BuildScriptOutput {
     }
 
     /// Convert a vector of [BuildScriptOutput] into a flagfile.
-    pub fn to_flags(v: &Vec<BuildScriptOutput>) -> String {
-        v.iter()
-            .filter_map(|x| match x {
-                BuildScriptOutput::Cfg(e) => Some(format!("--cfg={}", e)),
-                BuildScriptOutput::Flags(e) => Some(e.to_owned()),
-                BuildScriptOutput::LinkLib(e) => Some(format!("-l{}", e)),
-                BuildScriptOutput::LinkSearch(e) => Some(format!("-L{}", e)),
-                _ => None,
-            })
-            .collect::<Vec<_>>()
-            .join(" ")
+    pub fn to_flags(v: &Vec<BuildScriptOutput>, exec_root: &str) -> CompileAndLinkFlags {
+        let mut compile_flags = Vec::new();
+        let mut link_flags = Vec::new();
+
+        for flag in v {
+            match flag {
+                BuildScriptOutput::Cfg(e) => compile_flags.push(format!("--cfg={}", e)),
+                BuildScriptOutput::Flags(e) => compile_flags.push(e.to_owned()),
+                BuildScriptOutput::LinkLib(e) => link_flags.push(format!("-l{}", e)),
+                BuildScriptOutput::LinkSearch(e) => link_flags.push(format!("-L{}", e)),
+                _ => { },
+            }
+        }
+        CompileAndLinkFlags {
+            compile_flags: compile_flags.join(" "),
+            link_flags: link_flags.join(" ").replace(exec_root, "$EXEC_ROOT"),
```

Comment:
Add `{}` around the exec root
```suggestion
            link_flags: link_flags.join(" ").replace(exec_root, "${EXEC_ROOT}"),
```

It also needs to be adapted in the test.

Reply:
Done